JoonasHeinonen/escape-from-zordoom-prison
# Feature/weapon wheel functionality

### [Weapon wheel inventory and practicality](https://trello.com/c/wOIct3Bi/32-weapon-wheel-inventory-and-practicality)

https://user-images.githubusercontent.com/33086486/202927945-27e705e9-5675-4f38-846e-36b74839ed08.mov

**Changes:**
 - Added all inventory variables to `Globle.gd`  autoload script.
 - Created match statements to determine the position of a weapon in inventory and load the texture to it.
 - Created similar logic in changing the weapon used in the hand.
 - If no weapon is available, the weapon instance is set to `null`.
 - Created `WeaponCollectibles`. Weapon configurable into them. If a weapon is already in the array for currently held weapons, self-exterminate. If not, it's added to the player's inventory.
 - Player can now change the weapons.
 - Created functions for each weapon's shooting functionality, now prints how they should act, the only exception being Edge Blaster.
 - Changed the hierarchy of the nodes for the test level. **WARNING: Remember to keep the player as low as possible in the hierarchy order when creating levels.**

**How to test:**
 - You can add all the weapons to the test levels and change the variable of the weapon. If a weapon is already player owned.
 - Test player's inventory functionality. The key to switch the inventory on is `c`.
 - See that the code is useful software-wise and according to the standards.

**Sources used:**
 - [Exporting to a drop down list](https://godotengine.org/qa/63907/exporting-to-a-drop-down-list)
 - [Array (void *append*)](https://docs.godotengine.org/en/stable/classes/class_array.html)
 - [button not responding to mouse](https://godotengine.org/qa/62742/button-not-responding-to-mouse)
